apps/heft/src/plugins/TypeScriptPlugin/EmitFilesPatch.ts

```diff
@@ -191,6 +197,30 @@ export class EmitFilesPatch {
     };
   }
 
+  public static getRedirectedFilePath(filePath: string): string {
+    if (!EmitFilesPatch.isInstalled) {
+      throw new InternalError(
+        'EmitFilesPatch.getRedirectedFilePath() cannot be used unless the patch is installed'
+      );
+    }
+
+    // Redirect from "path/to/lib" --> "path/to/.heft/build-cache/lib"
+    let redirectedFilePath: string = filePath;
+    if (EmitFilesPatch._redirectedOutDir !== undefined) {
+      if (Path.isUnderOrEqual(filePath, EmitFilesPatch._originalOutDir!)) {
+        redirectedFilePath = path.resolve(
+          EmitFilesPatch._redirectedOutDir,
+          path.relative(EmitFilesPatch._originalOutDir!, filePath)
+        );
+      } else {
+        // The compiler is writing some other output, for example:
```

Comment:
Is this the only output that isn't written to the output directory? Seems likely. In which case, can we just check against that one file?

Reply:
Currently, I believe it is. But that may change in the future.